### PR TITLE
Safer variant updates in variantS controllers

### LINKED ISSUE #3857: Check `case_panels` stored on variant collection
`case_panels` were as far as I remember designed to be used in display populated versions of the variants, not to be stored back to db. This is not a huge issue, but variants are plentiful, and I don't quite see the need to save this permanently, when it anyway has to be rechecked each time the variant is to be displayed. I rather suspect I either made a mistake in that PR (#3837?) and/or that we have an older issue with variants getting re-written to db on load (perhaps like an overzealous re-check of compounds writing the whole in memory variant, rather than a select portion of it?).

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
Fix #3857

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

File: scout/server/blueprints/variants/controllers.py [hunk 1]
```diff
@@ -895,11 +872,18 @@ def parse_variant(
...
+            store.variant_update_field(
+                variant_id=variant_obj["_id"],
+                field_name="compounds",
+                field_value=variant_obj["compounds"],
+            )
 
     genes_have_changed = update_variant_genes(store, variant_obj, genome_build)
-
-    if update and (compounds_have_changed or genes_have_changed):
-        update_variant_store(store, variant_obj)
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/controllers.py:902]: This is the only place in scout where this function is invoked

File: scout/adapter/mongo/variant.py [hunk 2]
```diff
@@ -221,6 +222,14 @@ def count_variants(self, case_id, query, variant_ids, category):
         query = self.build_query(case_id, query=query, variant_ids=variant_ids, category=category)
         return self.variant_collection.count_documents(query)
 
+    def variant_update_field(self, variant_id: str, field_name: str, field_value: Any) -> dict:
```
Comment (dnil) [inline scout/adapter/mongo/variant.py:225]: 👍

File: scout/server/blueprints/variants/controllers.py [hunk 3]
```diff
@@ -513,7 +511,7 @@ def update_variant_genes(store, variant_obj, genome_build):
 
     if variant_genes is not None:
         for gene_obj in variant_genes:
-            # If there is no hgnc id there is nothin we can do
+            # If there is no hgnc id there is nothing we can do
```
Comment (dnil) [inline scout/server/blueprints/variants/controllers.py:514]: Jus sayin, aint sayin nuthin at all, jus sayin.

File: scout/server/blueprints/variants/controllers.py [hunk 4]
```diff
@@ -895,11 +872,18 @@ def parse_variant(
     compounds_have_changed = False
     if get_compounds:
         compounds_have_changed = update_compounds(store, variant_obj, case_dismissed_vars)
+        if update and compounds_have_changed:
```
Comment (dnil) [inline scout/server/blueprints/variants/controllers.py:875]: 💯

File: tests/adapter/mongo/test_variant_handling.py [hunk 5]
```diff
@@ -3,13 +3,28 @@
...
 import os
 
-import pytest
-
 TRAVIS = os.getenv("TRAVIS")
 
 LOG = logging.getLogger(__name__)
 
 
+def test_variant_update_field(real_variant_database):
```
Comment (dnil) [inline tests/adapter/mongo/test_variant_handling.py:11]: 🏅

**Review by dnil — APPROVED** [review]
Yes, much cleaner! Much approve.

**Pull request merged**